modules/analysis-common/src/test/java/org/elasticsearch/analysis/common/CompoundAnalysisTests.java

```diff
@@ -63,6 +67,44 @@ public void testDictionaryDecompounder() throws Exception {
         assertWarnings("Setting [version] on analysis component [custom7] has no effect and is deprecated");
     }
 
+    public void testHyphenationDecompoundingAnalyzerOnlyLongestMatch() throws Exception {
+        Settings[] settingsArr = new Settings[] { getJsonSettings(), getYamlSettings() };
+        for (Settings settings : settingsArr) {
+            List<String> terms = analyze(settings, "hyphenationDecompoundingAnalyzerOnlyLongestMatch", "kaffeemaschine fussballpumpe");
+            MatcherAssert.assertThat(
+                terms,
+                hasItems("kaffeemaschine", "kaffee", "fee", "maschine", "fussballpumpe", "fussball", "ballpumpe", "pumpe")
+            );
+        }
+        assertWarnings("Setting [version] on analysis component [custom7] has no effect and is deprecated");
+    }
+
+    /**
+     * Given a word list of: ["kaffee", "fee", "maschine"]
```

Comment:
is that the word list being used or is it this: [fuss, fussball, ballpumpe, ball, pumpe, kaffee, fee, maschine].  I was thrown off by the comment but had trouble tracking that through in my head.  Same thing on the comment on the subsequent test.  The test result makes sense to me and looks good.

Reply:
Technically, the wordlist contains `["fuss", "fussball", "ballpumpe", "ball", "pumpe", "kaffee", "fee", "maschine"]`, as defined in `test1.json:43`. The comment should highlight, that this parameter should solve this specific problem of preventing the match of "fee" (fairy) within "kaffee" (coffee). 

I left in the same wordlist for all tests and input text to ensure that they are not any unintended side effect. 

If it's clearer I could isolate the tests and only include the _**Kaffeemaschine**_ related words in this test and only the **_Fussballpumpe_** in the other one.

Reply:
gotcha I'm tracking now; this comment was a "for example".   So I'll just nit (change it if you want).  I'd just include before the comment something like "for example given a word list of: " ... that way it's clear that the test is validating more than just that word list.